render-wasm/src/wasm/fills.rs

```diff
@@ -53,6 +53,19 @@ impl TryFrom<&[u8]> for RawFillData {
     }
 }
 
+pub fn parse_fills_from_bytes(buffer: &[u8], num_fills: usize) -> Vec<shapes::Fill> {
+    let mut fills = Vec::new();
+
+    for i in 0..num_fills {
+        let start = i * RAW_FILL_DATA_SIZE;
+        let end = start + RAW_FILL_DATA_SIZE;
+        let bytes = &buffer[start..end];
+        let raw_fill = RawFillData::try_from(bytes).expect("Invalid fill data");
+        fills.push(raw_fill.into());
+    }
+    fills
+}
```

Comment:
`Vec::from_raw_parts` instead of iterating, and then we can map each entry with a `try_from`.